### PR TITLE
Generalized Strategy.Device test for all resources

### PR DESCRIPTION
**Pull request by jdahm**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
Neat! Does the `SizeLimiter` work well?

**Comment by jdahm** [comment]
Yeah, the SizeLimiter throws an exception when the threshold is reached, and the DynamicPool catches that and tries again. Very cool!

File: tests/integration/strategy_tests.cpp [hunk 1]
```diff
@@ -119,32 +119,21 @@ TEST_P(StrategyTest, Duplicate)
...
-TEST(Strategy, Device)
+TEST(DynamicPool, LimitedResource)
 {
   auto& rm = umpire::ResourceManager::getInstance();
 
-  auto allocator = rm.getAllocator("DEVICE");
-  void* alloc = nullptr;
+  const std::size_t max_mem{1024 * 1024 * 4};
 
-  // Determine how much memory we can allocate from device
```
Comment (mcfadden8) [inline tests/integration/strategy_tests.cpp]: Why are we throwing this out?

Comment (jdahm) [reply tests/integration/strategy_tests.cpp]: This part was integral for the test, to make the allocator `throw` internally further down. I replaced it by a `SizeLimiter` strategy, so we don't need this part any more.

**Comment by jdahm** [comment]
This is ready to go!

**Comment by davidbeckingsale** [comment]
@jdahm - was a JIRA ticket created w.r.t. testing for resource allocation failures?

**Comment by jdahm** [comment]
Yes, it's #469 on JIRA!

**Pull request merged**